apps/site/src/pages/account/api-keys-panel/types.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch, ReactElement, SetStateAction } from "react";
+
+import { UserFacingApiKeyProperties } from "../../../lib/api/model/api-key.model";
+
+export type ApiKeyProps = UserFacingApiKeyProperties;
+
+export type KeyAction = "rename" | "revoke";
+
+export type KeyActionStatus =
+  | {
+      publicId: string;
+      action: KeyAction;
+    }
+  | undefined;
+
+export interface ApiKeyItemProps {
+  renameApiKeyCard: ReactElement;
+  revokeApiKeyCard: ReactElement;
+  matchingNewlyCreatedKey?: string;
+  apiKey: ApiKeyProps;
+  keyAction?: KeyAction;
+}
+
+export interface ApiKeysContextValue {
+  apiKeys: ApiKeyProps[];
+  setApiKeys: Dispatch<SetStateAction<ApiKeyProps[]>>;
+  isCreatingNewKey: boolean;
+  setIsCreatingNewKey: Dispatch<SetStateAction<boolean>>;
+  newlyCreatedKeyIds: string[];
```

Comment:
I wonder whether `apiKeyProps` should just have a `new` boolean flag so we can get rid of this.

Reply:
This is used to keep track of the temporarily exposed `newly created` api keys - as full key values instead of only the public id part of it. I thought it'd make more sense to have those in a separate state, then check against that list to see if an api key is recently created.

The "storing the newly ones in a separate state" is pretty useful imo, because right now we *refect* them from backend when updates happen.

Reply:
To be honest, I'm still a bit uneasy about storing these in a separate array instead of as part of the key, but I don't think it's worth blocking a PR for. Happy to leave this.